### PR TITLE
[MM-794]: Fixed the issue of fields not prefilled when creating issue with custom fields

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Fixed the issue of fields not being prefilled when creating issues with custom fields. Unable to support the fields related to the users because the Username is not available in the create issue data and all the user picker fields on the Jira create issue form work on the username.
##### Added Support for: 
- Labels
- Select list(multiple choices)
- Checkboxes
#####  Fields still not supported
- User picker
- Assignee
- Reporter

### Video Link
https://flonnect.com/video/6d26327abd0d-4a7d-855a-3f4c909e8648

### How to test
- On a Jira instance create the following custom fields Date Picker, Date Time Picker, Number Field
- Configure the fields to be mandatory for the Screen / Project / Issue type you are using
- Attempt to create an issue through Mattermost
- Provide values for all fields
- Create the issue
- Follow the link

### CODE REVIEW

File: server/client.go [hunk 1]
```diff
@@ -334,14 +334,31 @@ func MakeCreateIssueURL(instance Instance, project *jira.Project, issue *jira.Is
...
+			q.Add(k, vTyped)
+		case map[string]interface{}:
+			if id, ok := vTyped["id"].(string); ok {
 				q.Add(k, id)
 			}
+		case []interface{}:
+			for _, item1 := range vTyped {
+				switch vTyped2 := item1.(type) {
+				case string:
+
```
Comment (raghavaggarwal2308) [inline server/client.go]:
```suggestion
```

File: server/client.go [hunk 2]
```diff
@@ -334,14 +334,31 @@ func MakeCreateIssueURL(instance Instance, project *jira.Project, issue *jira.Is
...
-			if id, ok := mapV["id"].(string); ok {
+		switch vTyped := v.(type) {
+		case string:
+			q.Add(k, vTyped)
+		case map[string]interface{}:
+			if id, ok := vTyped["id"].(string); ok {
 				q.Add(k, id)
 			}
+		case []interface{}:
+			for _, item1 := range vTyped {
```
Comment (raghavaggarwal2308) [inline server/client.go]: Use meaningful names for item1, vtyped1 etc

File: server/client.go [hunk 3]
```diff
@@ -334,14 +334,31 @@ func MakeCreateIssueURL(instance Instance, project *jira.Project, issue *jira.Is
 
 	// add custom fields
 	for k, v := range issue.Fields.Unknowns {
-		strV, ok := v.(string)
-		if ok {
-			q.Add(k, strV)
-		}
-		if mapV, ok := v.(map[string]interface{}); ok {
-			if id, ok := mapV["id"].(string); ok {
+		switch vTyped := v.(type) {
```
Comment (raghavaggarwal2308) [inline server/client.go]: Are we handling all the cases here now? If not mention the same in the description with proper explanation.

**Review by raghavaggarwal2308 — COMMENTED** [review]
Add a video showing the demo for the fix in description as this is complex case. This will ensure that there is no objection and clarifications left for us

**Comment by Kshitij-Katiyar** [comment]
> Add a video showing the demo for the fix in description as this is complex case. This will ensure that there is no objection and clarifications left for us

Will do

**Comment by Kshitij-Katiyar** [comment]
> Add a video showing the demo for the fix in description as this is complex case. This will ensure that there is no objection and clarifications left for us

Done

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR on mattermost remote with all these changes hence closing this one

**Pull request closed without merging**